cdap-proto/src/main/java/co/cask/cdap/proto/RunRecord.java

```diff
@@ -131,6 +136,10 @@ public ProgramRunCluster getCluster() {
     return cluster == null ? new ProgramRunCluster(ProgramRunClusterStatus.DEPROVISIONED, null, null) : cluster;
   }
 
+  public ProfileId getProfileId() {
+    return profileId;
```

Comment:
for backwards compatibility with run records prior to 5.0.0, we should return the native profile if profileId is null.